tools/scalar-schema/src/scalar_schema/jdbc.clj

```diff
@@ -0,0 +1,298 @@
+       (if key-type (str "'" key-type "'") "NULL") ","
+       (if key-order (str "'" key-order "'") "NULL") ","
+       secondary-indexed ","
+       (if index-order (str "'" index-order "'") "NULL") ","
+       ordinal-position
+       ")"))
+
+(defn- get-key-type
+  [column partition-key clustering-key]
+  (cond
+    (key? column partition-key) "PARTITION"
+    (key? column clustering-key) "CLUSTERING"
+    :else nil))
+
+(defn- get-secondary-indexed [rdb-engine column secondary-index]
+  (if (key? column secondary-index) (boolean-value-true rdb-engine)
+      (boolean-value-false rdb-engine)))
+
+(defn- zipseq
+  ([coll] (partition 1 coll))
+  ([c1 & colls] (let [c (cons c1 colls)] (partition (count c) (apply interleave c)))))
+
+(defn- insert-metadata
+  [db-spec rdb-engine prefix schema table partition-key clustering-key columns secondary-index]
+  (doall (map #(let [column (first (first %))
+                     data-type (string/upper-case (second (first %)))
+                     key-type (get-key-type column partition-key clustering-key)
+                     key-order (if (key? column clustering-key) "ASC" nil)
+                     secondary-indexed (get-secondary-indexed rdb-engine column secondary-index)
+                     index-order (if (key? column secondary-index) "ASC" nil)
+                     ordinal-position (second %)]
+                 (execute! db-spec (make-insert-metadata-statement rdb-engine prefix schema table
+                                                                   column data-type key-type
+                                                                   key-order secondary-indexed
+                                                                   index-order ordinal-position)))
+              (zipseq columns (iterate inc 1)))))
+
+(defn- make-drop-table-statement
+  [rdb-engine schema table]
+  (str "DROP TABLE " (get-table-name rdb-engine schema table)))
+
+(defn- drop-table
+  [db-spec rdb-engine schema table]
+  (execute! db-spec (make-drop-table-statement rdb-engine schema table)))
+
+(defn- make-delete-metadata-statement
+  [rdb-engine prefix schema table]
+  (let [full_table_name (get-table-name schema table)]
+    (str "DELETE FROM "
+         (get-metadata-table-name rdb-engine prefix)
+         " WHERE " (enclose rdb-engine "full_table_name") " = "
+         "'" full_table_name "'")))
+
+(defn- delete-metadata
+  [db-spec rdb-engine prefix schema table]
+  (execute! db-spec (make-delete-metadata-statement rdb-engine prefix schema table)))
+
+(defn- reorder-columns
+  [columns partition-key clustering-key]
+  (concat (map #(vector % (get columns %)) partition-key)
+          (map #(vector % (get columns %)) clustering-key)
+          (filter #(not (key? (first %) (concat partition-key clustering-key)))
+                  (apply list columns))))
+
+(defn make-jdbc-operator
+  [{:keys [jdbc-url user password prefix]}]
+  (let [db-spec (get-db-spec jdbc-url user password)
```

Comment:
I think we can make an executor to execute SQL statements.
```clojure
(defn- make-execution-fn
  [{:keys [jdbc-url user password]}]
  (fn [statement]
    (-> {:connection-uri jdbc-url :user user :password password}
        (jdbc/execute! [statement]))))
...
(defn make-jdbc-operator
  [{:keys [jdbc-url user password prefix] :as opts}]
  (let [execution-fn (get-executor opts)
        rdb-engine (get-rdb-engine opts)
        enclosure-fn (get-enclosure rdb-engine)]
...
    (create-table executor enclosure schema opts)
...
```

It might be better to add them to `opts` like this:
```clojure
(defn make-jdbc-operator
  [opts]
  (let [opts (assoc opts
                    :execution-fn (get-execution-fn opts)
                    :rdb-engine (get-rdb-engine opts)
                    :enclosure-fn (get-enclosure-fn rdb-engine))]
    (create-metadata-schema opts)
```